homeassistant/components/bluetooth/__init__.py

```diff
@@ -0,0 +1,224 @@
+
+import asyncio
+from collections.abc import Awaitable, Callable
+import dataclasses
+from enum import Enum
+import logging
+from typing import Any
+
+from bleak import BleakError
+from bleak.backends.device import MANUFACTURERS, BLEDevice
+from bleak.backends.scanner import AdvertisementData
+
+# from homeassistant.components import websocket_api
+# from homeassistant.components.websocket_api.connection import ActiveConnection
+from homeassistant.const import EVENT_HOMEASSISTANT_STARTED, EVENT_HOMEASSISTANT_STOP
+from homeassistant.core import (
+    CALLBACK_TYPE,
+    Event,
+    HomeAssistant,
+    callback as hass_callback,
+)
+from homeassistant.data_entry_flow import BaseServiceInfo
+
+# from homeassistant.helpers import discovery_flow, system_info
+# from homeassistant.helpers.debounce import Debouncer
+from homeassistant.helpers.typing import ConfigType
+from homeassistant.loader import bind_hass
+
+from . import models
+from .const import DOMAIN
+from .models import HaBleakScanner
+from .usage import install_multiple_bleak_catcher
+
+# import voluptuous as vol
+
+# from homeassistant.loader import async_get_bluetooth
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class BluetoothScanningMode(Enum):
+    """The mode of scanning for bluetooth devices."""
+
+    PASSIVE = "passive"
+    ACTIVE = "active"
+
+
+SCANNING_MODE_TO_BLEAK = {
+    BluetoothScanningMode.ACTIVE: "active",
+    BluetoothScanningMode.PASSIVE: "passive",
+}
+
+
+@dataclasses.dataclass
+class BluetoothServiceInfo(BaseServiceInfo):
+    """Prepared info from bluetooth entries."""
+
+    name: str
+    address: str
+    rssi: int
+    manufacturer_data: dict[int, str]
+    service_data: dict[str, bytes]
+    service_uuids: list[str]
+
+    @classmethod
+    def from_advertisement(
+        cls, device: BLEDevice, advertisement_data: AdvertisementData
+    ) -> BluetoothServiceInfo:
+        """Create a BluetoothServiceInfo from an advertisement."""
+        return cls(
+            name=advertisement_data.local_name or device.name or device.address,
+            address=device.address,
+            rssi=device.rssi,
+            manufacturer_data=advertisement_data.manufacturer_data,
+            service_data=advertisement_data.service_data,
+            service_uuids=advertisement_data.service_uuids,
+        )
+
+    @property
+    def manufacturer(self) -> str | None:
+        """Convert manufacturer data to a string."""
+        for manufacturer in self.manufacturer_data:
+            if name := MANUFACTURERS.get(manufacturer):
+                return name
+        return None
+
+    @property
+    def manufacturer_id(self) -> int | None:
+        """Get the first manufacturer id."""
+        for manufacturer in self.manufacturer_data:
+            return manufacturer
+        return None
+
+
+BluetoothChange = Enum("BluetoothChange", "ADVERTISEMENT")
+BluetoothCallback = Callable[[BluetoothServiceInfo, BluetoothChange], Awaitable]
+
+
+@bind_hass
```

Comment:
bind_hass is legacy.

Reply:
4c5f68dc5965fc08565865c65725e437a90caf17